tests/default/_core/search/knn_disk_based.yaml

```diff
@@ -0,0 +1,71 @@
+$schema: ../../../../json_schemas/test_story.schema.yaml
```

Comment:
Let's make a folder for these and match the relevant option name, so `knn/on_disc.yaml`.